flask_security/twofactor.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+"""
+    flask_security.two_factor
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Flask-Security two_factor module
+
+    :copyright: (c) 2016 by Gal Stainfeld, at Emedgene
+"""
+
+import os
+import base64
+
+import pyqrcode
+import onetimepass
+from flask import current_app as app, session, abort
+from flask_login import current_user
+from werkzeug.local import LocalProxy
+
+from .utils import send_mail, config_value, get_message, do_flash,\
+    SmsSenderFactory, login_user
+from .signals import user_two_factored, two_factor_method_changed
+
+# Convenient references
+_security = LocalProxy(lambda: app.extensions['security'])
+
+_datastore = LocalProxy(lambda: _security.datastore)
+
+
+def send_security_token(user, method, totp_secret):
+    """Sends the security token via email for the specified user.
+    :param user: The user to send the code to
+    :param method: The method in which the code will be sent
+                ('mail' or 'sms') at the moment
+    :param totp_secret: a unique shared secret of the user
+    """
+    token_to_be_sent = get_totp_password(totp_secret)
+    if method == 'mail':
+        send_mail(config_value('EMAIL_SUBJECT_TWO_FACTOR'),
+                  user.email,
+                  'two_factor_instructions',
+                  user=user,
+                  token=token_to_be_sent)
+    elif method == 'sms':
+        msg = "Use this code to log in: %s" % token_to_be_sent
+        from_number = config_value('TWO_FACTOR_SMS_SERVICE_CONFIG')[
+            'PHONE_NUMBER']
+        if 'phone_number' in session:
+            to_number = session['phone_number']
+        else:
+            to_number = user.phone_number
+        sms_sender = SmsSenderFactory.createSender(
+            config_value('TWO_FACTOR_SMS_SERVICE'))
+        sms_sender.send_sms(from_number=from_number,
+                            to_number=to_number, msg=msg)
+
+    elif method == 'google_authenticator':
+        # password are generated automatically in the google authenticator app
+        pass
+
+
+def get_totp_uri(username, totp_secret):
```

Comment:
Question - did you consider using passlib.totp classes rather than rolling your own?